cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```diff
@@ -0,0 +1,248 @@
+        parser.add_argument('--user',
+                            dest='user_email',
+                            default=False,
+                            required=True,
+                            help=self.user_help)
+        parser.add_argument('--course',
+                            dest='course_id',
+                            default=False,
+                            required=False,
+                            help=self.course_help)
+        parser.add_argument('--force-update-catalog-visibility-about',
+                            dest='force_update_catalog_visibility_about',
+                            action='store_true',
+                            help=self.force_update_catalog_visibility_about_help)
+
+        # WARNING
+        # Limiting the all courses bulk advanced setting update here to avoid consuming all
+        # server memory. Defaulting bulk operations to 150 courses but you can use this argument
+        # to specify any value different than the default.
+        parser.add_argument('--limit-bulk-update-operation',
+                            dest='limit_bulk_update_operation',
+                            default=150,
+                            required=False,
+                            help=self.limit_bulk_update_operation_help)
+
+    def handle(self, *args, **options):
+        """
+        Execute the command
+        """
+        self._force_update_catalog_visibility_about = options.get(
+            'force_update_catalog_visibility_about', False
+        )
+        try:
+            update_fields = json.loads(options["fields_json"])
+        except ValueError as err:
+            raise CommandError("Invalid JSON object") from err
+
+        # Update course(s) to push out `fields_json` of advanced setting changes.
+        self._set_courses_advanced_fields(
+            update_fields,
+            options['user_email'],
+            options['course_id'],
+            int(options['limit_bulk_update_operation'])
+        )
+
+    def _set_courses_advanced_fields(self, update_fields, user_email, course_id, limit_bulk_update_operation):  # pylint: disable=line-too-long
+        """
+        Export all courses to target directory and return the list of courses which failed to export
+        """
+
+        try:
+            user = user_from_str(user_email)
+        except User.DoesNotExist:
+            logger.warning(user_email + " user does not exist")  # lint-amnesty, pylint: disable=logging-not-lazy
+            logger.warning("Can't update fields")
+            return
+
+        # Single Course
+        if course_id:
+            course_module = modulestore().get_course(CourseKey.from_string(course_id))
+
+            revised_update_fields = self._force_catalog_visibility_about(
+                update_fields, course_module
+            )
+
+            print(f"Applying revised_update_fields = {revised_update_fields} to course {course_id} using `{user}` user.\n")  # pylint: disable=line-too-long
```

Comment:
Don't use `print`. Instead, use `logger.info` or `logger.debug`. Add a `-v/--verbose` option. Adjust the logger log level based on that option.